packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -292,3 +296,49 @@ export async function reduceInBatchesSerially<
   const finalResult = workingResult as Result;
   return finalResult;
 }
+
+/**
+ * fetches and maps the exchange rates
+ * @param nativeCurrency - native currency
+ * @param tokenAddresses - array of token addresses
+ * @param chainId - The ID of the chain.
+ */
+export async function fetchAndMapExchangeRates(
+  nativeCurrency: string,
+  tokenAddresses: Hex[],
+  chainId: Hex,
+): Promise<ContractExchangeRates> {
+  const tokenPricesFetchService: AbstractTokenPricesService =
```

Comment:
TokenRatesController takes a service object to fetch exchange rates so that clients can choose which one they would like to use; if they want to substitute that object out for another — for instance because we want to switch to a different API — they can easily do so without us having to make a change to this package. It seems that we would now be hardcoding the Price API into this function. Perhaps we could have this function take a `tokenPricesService` instead?

Also, with the addition of a fourth argument, it might be good to change this function to take an object rather than a series of positional arguments. I worry that when this function is called it may be difficult to understand what the arguments are — but an object will make that clear as the property names will spell that out.

Reply:
Sorry, another suggestion: The name of this concept is "token prices service", not "token prices fetch service", so I think we should be consistent rather than use two names. What do you think about using `tokenPricesService` as the name of the variable?

Reply:
Thanks for the feedback! 🙏  totally agree that it should take the tokenPricesService as an arg instead; will update the fct!